common/persistence/cassandra/cassandraHistoryPersistence.go

```diff
@@ -176,11 +176,11 @@ func (h *nosqlHistoryManager) ReadHistoryBranch(
 
 		switch {
 		case nodeID < lastNodeID:
-			return nil, &types.InternalServiceError{
+			return nil, &types.InternalDataInconsistencyError{
```

Comment:
There's some more checks in `historyStore.go` regarding corrupted history. Can we also return this inconsistency error if those checks fail?

Reply:
Updated those errors in the history store. I think there might be more cases to be converted. I don't plan to convert all of them in this change. We can continue with other changes.

Reply:
Sure thing. I method those in historyStore because I saw them a lot in production logs. We can gradually convert the rest.